ios/FluentUI/Vnext/PersonaButton/PersonaButton.swift

```diff
@@ -211,22 +213,44 @@ public struct PersonaButton: View {
         self.tokens = state.tokens
     }
 
+    private var personaText: some View {
+        Group {
+            Text(state.primaryText ?? "")
+                .frame(maxWidth: .infinity, alignment: .center)
+                .scalableFont(font: tokens.labelFont)
+                .foregroundColor(Color(tokens.labelColor))
+            if state.buttonSize.shouldShowSubtitle {
+                Text(state.secondaryText ?? "")
+                    .frame(maxWidth: .infinity, alignment: .center)
+                    .scalableFont(font: tokens.sublabelFont)
+                    .foregroundColor(Color(tokens.sublabelColor))
+            }
+        }
+        .frame(width: buttonMaxWidth)
+    }
+
+    private var avatarView: some View {
+        AvatarView(avatarState)
+            .padding(.top, tokens.padding)
+            .padding(.bottom, tokens.avatarInterspace)
+            .padding(.horizontal, tokens.padding)
+            .background(GeometryReader { geometry in
+                Color.clear.preference(
```

Comment:
Do you need the preference key to achieve this?
Why not just set the maxWidth of the entire frame to the property and apply the padding to it?

Reply:
Originally I was trying to tie this to the width of the Avatar, so that if the design picks a new `AvatarSize` then this would update automatically. But looking at it with a fresh set of eyes, I think that might be over-engineered. I'm gonna switch it to a design token and simplify all this code.